addon/devtoolsPanel.js

```diff
@@ -1,47 +1,232 @@
-                                options: {shouldScroll: false, wait: 0}});
+// Draw the table with the list of existing labeled elements.
+// If an element is inspected/selected with devtools (i.e. $0 is defined), it
+// will always be the first row in the table.
+function updateLabeledTable(labeled) {
+
+    function addTextCell(row, text, isHeader=false) {
+        const td = document.createElement(isHeader ? 'th' : 'td');
+        td.appendChild(document.createTextNode(text));
+        row.appendChild(td);
+        return td;
+    }
+
+    function inspectLabeled(event) {
+        event.preventDefault();
+        const js = 'inspect(document.querySelector("' + event.target.dataset.path + '"))';
+        browser.devtools.inspectedWindow.eval(js)
+            .catch((e) => {
+                console.error(e);
+            });
+    }
+
+    function addPathCell(row, label) {
+        const td = document.createElement('td');
+        td.classList.add('path');
+        if (label.inspected) {
+            td.appendChild(document.createTextNode(label.path));
+        } else {
+            let a = document.createElement('a');
+            a.href = '#';
+            a.dataset.path = label.path;
+            a.addEventListener('click', inspectLabeled);
+            a.appendChild(document.createTextNode(label.path));
+            td.appendChild(a);
+        }
+        row.appendChild(td);
+    }
+
+    function updateLabel(event) {
+        backgroundPort.postMessage({
+            type: 'label',
+            tabId: browser.devtools.inspectedWindow.tabId,
+            selector: event.target.dataset.path,
+            label: event.target.value.trim(),
+        });
+    }
+
+    function addLabelCell(row, label) {
+        const td = document.createElement('td');
+        td.classList.add('label');
+        const input = document.createElement('input');
+        input.setAttribute('placeholder', 'unlabeled');
+        input.value = label.label;
+        input.dataset.path = label.path;
+        input.addEventListener('change', updateLabel);
+        td.appendChild(input);
+        row.appendChild(td);
+    }
+
+    function removeLabeled(event) {
+        event.preventDefault();
+        backgroundPort.postMessage({
+            type: 'label',
+            tabId: browser.devtools.inspectedWindow.tabId,
+            selector: event.target.dataset.path,
+            label: '',
+        });
+    }
+
+    function addClearLabelCell(row, label) {
+        const td = document.createElement('td');
+        td.classList.add('action');
+        const a = document.createElement('a');
+        a.href = '#';
+        a.classList.add('action-button');
+        a.dataset.path = label.path;
+        a.dataset.label = label.label;
+        a.setAttribute('title', label.inspected ? 'Clear Label' : 'Remove Label');
+        a.addEventListener('click', removeLabeled);
+        a.appendChild(document.createTextNode(label.inspected ? '⦸' : '✖'));
```

Comment:
I find the ⦸ button more confusing than anything. Can we lose it? I'd leave it for now, except it also doesn't perceive as working reliably: when your insertion point is in the label field and you click ⦸, it only defocuses the field; it doesn't actually activate the button. At least, that's in 61.0.1.

Reply:
yup; will drop that button